e2e/scenario_helpers_test.go

```diff
@@ -126,6 +126,9 @@ func createAndValidateVM(ctx context.Context, t *testing.T, scenario *Scenario)
 		validateWasm(ctx, t, scenario.Runtime.Cluster.Kube, nodeName)
 	}
 
+	if scenario.Tags.KubeletCustomConfig {
```

Comment:
yes and no. i have introduced the tag and have only used it in 2 e2e scenarios so far

https://github.com/Azure/AgentBaker/blob/979c12f5af266491d18b7d8ac2bc0d8b9e09d87e/e2e/scenario_test.go#L1258

i did not use tag "SeccompDefaultProfile" ish tag for the reason that this is only one of the flags we used in customKubeletConfig. since this is e2e, i think customKubeletConfig tag is at a better abstraction level.

leaving space for other customKubeletConfig checks when we want to include in retrospect and to consolidate the e2e test scenarios maybe around VHD * OS SKU matrix .

Reply:
I think this is special case for a single case. I would instead add another hook to `Scenario`.

Something like `DoExtraStuff: func(s *Scenario, t *testing.T)` or `PrepareCluster` (it's all bad names, come up with something better)

Reply:
alternatively, can it be a part of validator?